modules/caddyhttp/server.go

```diff
@@ -467,6 +468,12 @@ func (slc ServerLogConfig) getLoggerName(host string) string {
 	if loggerName, ok := slc.LoggerNames[host]; ok {
 		return loggerName
 	}
+	// Try matching wildcard domains if other non-specific loggers exist
+	for potentialHost, loggerName := range slc.LoggerNames {
+		if certmagic.MatchWildcard(host, potentialHost) {
```

Comment:
This extra loop can be avoided if we borrow some of the logic from the `MatchWildcard()` function. If we split `host` by its labels and then replace each next-left-most consecutive label with a `*` all we have to do is the map lookup. Does that make sense?

Not a big deal, but an optimization that should reduce the complexity a notch.

Reply:
I *think* I understand. So if my list of loggers is `domain.com`, `subdomain1.domain.com`, and `*.domain.com` and I'm looking up `something-else.domain.com` I can replace the leftmost label `something-else` with `*` and try the map lookup again with `*.domain.com`?

Reply:
Yep! It'll somewhat resemble `MatchWildcard` I think. (See its implementation.)

Reply:
Yep I think I get it. I just haven't ever had a use case (at home) where I was doing wildcard as segments of a subdomain like `ads.*.domain.com` and the like.